### PR TITLE
Remove permutation of KV cache 

### PR DESCRIPTION
**Pull request by harsh-nod**
This PR removes the permutation of the key and
value cache prior to kernel invocation. In order to accomplish this, we add an additional attribute to the MMA operator that specifies which dimensions map to the M, N, K dimensions of the matrix multiplication.

### CODE REVIEW

File: tests/kernel/wave/attention/paged_attention_test.py [hunk 1]
```diff
@@ -204,12 +204,8 @@ def testPagedFlashDecoding(
...
         head_size_kv = value_cache.shape[3]
 
-    permuted_key_cache = key_cache.view(num_seqs, -1, num_kv_heads, head_size).permute(
-        [0, 2, 1, 3]
-    )
-    permuted_value_cache = value_cache.view(
-        num_seqs, -1, num_kv_heads, head_size_kv
-    ).permute([0, 2, 3, 1])
+    permuted_key_cache = key_cache.view(num_seqs, -1, num_kv_heads, head_size)
+    permuted_value_cache = value_cache.view(num_seqs, -1, num_kv_heads, head_size_kv)
```
Comment (GMNGeoffrey) [inline tests/kernel/wave/attention/paged_attention_test.py:208]: No longer permuted, so change the variable names?

**Review by GMNGeoffrey — COMMENTED** [review]
Maybe add a test that demonstrates usage of this? It looks like it's `None` in all the lit tests still

File: lit_tests/kernel/wave/expansion.py [hunk 2]
```diff
@@ -309,21 +309,21 @@ def test_gemm():
         # CHECK-SAME: (%b, 4, None, (), None)
 
         # CHECK-NEXT: %mma_M:0_N:0_K:0
-        # CHECK-SAME: (%read_M:0_N:0_K:0, %read_M:0_N:0_K:0, %acc_M:0_N:0_K:0, None)
+        # CHECK-SAME: (%read_M:0_N:0_K:0, %read_M:0_N:0_K:0, %acc_M:0_N:0_K:0, None, None)
```
[the remark this answers is not in the document]
Comment (harsh-nod) [reply lit_tests/kernel/wave/expansion.py:312]: Doesn't the custom printing print the indices? (I don't think it print the vector shapes but we can add that as well)

File: tests/kernel/wave/attention/paged_attention_test.py (hunk 1 above)
Comment (harsh-nod) [reply tests/kernel/wave/attention/paged_attention_test.py:208]: Will do thank you

**Comment by harsh-nod** [comment]
Turns out I didnt have to introduce a new dimensional mapping construct, this is already handled by mapping.

File: lit_tests/kernel/wave/expansion.py (hunk 2 above)
[the remark this answers is not in the document]
Comment (GMNGeoffrey) [reply lit_tests/kernel/wave/expansion.py]: I only just found out the custom printing exists when writing the lit tests for my PR :-D I had been doing some of my own modifications on graph printing as well that I might try to contribute.

**Comment by harsh-nod** [comment]
Have opened a new PR for this. closing this out.
